frontend/src/components/GoalsTable/GoalsTableHeader.js

```diff
@@ -0,0 +1,112 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+import Pagination from 'react-js-pagination';
+import { Link } from 'react-router-dom';
+import UserContext from '../../UserContext';
+import { canEditOrCreateGoals } from '../../permissions';
+import { DECIMAL_BASE } from '../../Constants';
+
+export function renderTotal(offset, perPage, activePage, count) {
+  const from = offset >= count ? 0 : offset + 1;
+  const offsetTo = perPage * activePage;
+  let to;
+  if (offsetTo > count) {
+    to = count;
+  } else {
+    to = offsetTo;
+  }
+  return `${from}-${to} of ${count}`;
+}
+
+export default function GoalsTableHeader({
+  title,
+  count,
+  activePage,
+  offset,
+  perPage,
+  handlePageChange,
+  hidePagination,
+  recipientId,
+  regionId,
+  hasActiveGrants,
+}) {
+  const { user } = useContext(UserContext);
+  const hasButtonPermissions = canEditOrCreateGoals(user, parseInt(regionId, DECIMAL_BASE));
+
+  return (
+    <div className="desktop:display-flex">
+      <div className="desktop:display-flex flex-1 desktop:padding-top-0 padding-top-2">
+        <h2 className="font-body-lg margin-left-2 margin-right-1 margin-y-3">{title}</h2>
+        { hasActiveGrants && hasButtonPermissions ? (
+          <span className="smart-hub--table-controls desktop:margin-x-2 desktop:margin-y-0 margin-2 display-flex flex-row flex-align-center">
```

Comment:
We should create a new feature flag for creation of goals since we don't want goal creation to release at the same time as view goals

Reply:
I didn't realize that was the case... I wonder if we could just use the existing feature flag and the wrapping around the parts as we release them.

Reply:
either way, if its all right with you, I'm getting ready to merge this in and will write another Jira/PR to make some of more recent changes to the mockup and can include the feature flag on that one if needed

Reply:
yeah we could probably use the same flag, either way up to you. And putting it in the next PR is fine.